### PR TITLE
Import bams from iRODS and convert to fastqs.

### PR DESCRIPTION
**Pull request by CraigPorter**
This is a pipeline that I set up for the pathogens group which imports bams from IRODS, converts them to fastqs, deletes the bams then updates the tracking database. 

The module is based on VertRes::Pipelines::Import_iRODS and uses the get_bams() method from that module. It adapts a routine from VertRes::Pipelines::Mapping.pm to convert fastqs into bams using the bam2fastq() method in VertRes::Utils::Sam. Bam files are deleted once fastq files have been written. Finally, the tracking database is updated using the method from VertRes::Pipelines::Import.

Basically, I tried to avoid introducing redundant code. 

I was also asked to set the pipeline to skip a lane (without updating the database) if there were any non-bam files. I didn't want to skip thru the stages of the pipeline or leave any skip-this-lane files on the disk so I skipped running the pipeline by clearing the actions list. This method works but it just feels like a hacky way of doing things - any suggestions on an alternative method are welcome.

### CODE REVIEW

**Comment by tk2** [comment]
On 28/04/2011 17:09, CraigPorter wrote:

> This is a pipeline that I set up for the pathogens group which imports bams from IRODS, converts them to fastqs, deletes the bams then updates the tracking database.
> 
> The module is based on VertRes::Pipelines::Import_iRODS and uses the get_bams() method from that module. It adapts a routine from VertRes::Pipelines::Mapping.pm to convert fastqs into bams using the bam2fastq() method in VertRes::Utils::Sam. Bam files are deleted once fastq files have been written. Finally, the tracking database is updated using the method from VertRes::Pipelines::Import.
> 
> Basically, I tried to avoid introducing redundant code.
> 
> I was also asked to set the pipeline to skip a lane (without updating the database) if there were any non-bam files. I didn't want to skip thru the stages of the pipeline or leave any skip-this-lane files on the disk so I skipped running the pipeline by clearing the actions list. This method works but it just feels like a hacky way of doing things - any suggestions on an alternative method are welcome.
> 
> Excellent - it just so happens that I need this pipeline as we got a recent free test run that didnt have a reference sequence set so its in unmapped BAM format. Do you happen to have an example config file for this pipeline?

Thomas

## 

 The Wellcome Trust Sanger Institute is operated by Genome Research 
 Limited, a charity registered in England with number 1021457 and a 
 company registered in England with number 2742969, whose registered 
 office is 215 Euston Road, London, NW1 2BE.

**Comment by CraigPorter** [comment]
On 2 May 2011, at 17:53, tk2 wrote:

> On 28/04/2011 17:09, CraigPorter wrote:
> 
> > This is a pipeline that I set up for the pathogens group which imports bams from IRODS, converts them to fastqs, deletes the bams then updates the tracking database.
> > 
> > The module is based on VertRes::Pipelines::Import_iRODS and uses the get_bams() method from that module. It adapts a routine from VertRes::Pipelines::Mapping.pm to convert fastqs into bams using the bam2fastq() method in VertRes::Utils::Sam. Bam files are deleted once fastq files have been written. Finally, the tracking database is updated using the method from VertRes::Pipelines::Import.
> > 
> > Basically, I tried to avoid introducing redundant code.
> > 
> > I was also asked to set the pipeline to skip a lane (without updating the database) if there were any non-bam files. I didn't want to skip thru the stages of the pipeline or leave any skip-this-lane files on the disk so I skipped running the pipeline by clearing the actions list. This method works but it just feels like a hacky way of doing things - any suggestions on an alternative method are welcome.
> > 
> > Excellent - it just so happens that I need this pipeline as we got a recent free test run that didnt have a reference sequence set so its in unmapped BAM format. Do you happen to have an example config file for this pipeline?
> 
> Thomas

 Hi Thomas,

Sorry for late reply - I didn't check my email over the weekend.

There's nothing special about the conf file, it's just like the standard import conf file. Have a global config file to get the jobs to import and the filename of the config file. For example: 

**VRTrack_Import** importirods.conf

The config file would look something like this: 

root    => '/abs/path/to/root/data/dir',
module  => 'VertRes::Pipelines::Import_iRODS_fastq',
prefix  => '_',

db =>
   {
        database => 'pathogen_example_track',
        host     => 'web-mii-shap',
        port     => 3303,
        user     => 'pathpipe_ro',
        password => '',
   },
data => 
   {  # Pipeline-specific data
        db  => 
            {
               database => 'pathogen_example_track',
               host     => 'web-mii-shap',
               port     => 3303,
               user     => 'pathpipe_rw',
               password => 'xxx',  
            },
   }, 

The one (very minor) requirement is that the pipeline uses the iRODS module so you have to be in an iRODS session to use it however, (as you've been importing from iRODS) just running the pipeline the same way that you run the Import_iRODS pipeline will work.

Great to know that someone outside the Pathogens group wants to use the pipeline. If there's anything you need - just let me know.

Cheers, 

Craig

## 

 The Wellcome Trust Sanger Institute is operated by Genome Research 
 Limited, a charity registered in England with number 1021457 and a 
 company registered in England with number 2742969, whose registered 
 office is 215 Euston Road, London, NW1 2BE.
